core/app/models/spree/variant.rb

```diff
@@ -304,8 +304,13 @@ def should_track_inventory?
       self.track_inventory? && Spree::Config.track_inventory_levels
     end
 
+    # Image that can be used for the variant.
+    #
+    # Will first search for images on the variant, then all images from the
+    # products' variants. If all else fails, will return a new image object.
+    # @return [Spree::Image] the image to display
     def display_image
-      images.first || Spree::Image.new
+      images.first || product.variant_images.first || Spree::Image.new
```

Comment:
Based on what we were [originally doing](https://github.com/solidusio/solidus/blob/master/core/app/helpers/spree/base_helper.rb#L194-L206), I believe it's what we want. Simplifying the existing code for the case when we're looking for a variant's image:

```ruby
# Assume self == variant
if images.empty?
  if false
    # First part never going to happen, because we have a variant.
  else
    if !product.variant_images.empty?
      product.variant_images.first
    else
      # No image
    end
  else
    images.first
  end
end
```